packages/hydrogen/src/storefront.ts

```diff
@@ -259,11 +253,11 @@ export function createStorefrontClient<TI18n extends I18nBase>(
 
   const defaultHeaders = getHeaders({
     contentType: 'json',
-    buyerIp: storefrontHeaders?.buyerIp || buyerIp,
+    buyerIp: storefrontHeaders?.buyerIp || '',
   });
 
   defaultHeaders[STOREFRONT_REQUEST_GROUP_ID_HEADER] =
-    storefrontHeaders?.requestGroupId || requestGroupId || generateUUID();
+    storefrontHeaders?.requestGroupId || generateUUID();
```

Comment:
Unrelated but curious: does the SFAPI require this request-group-id ? Or could it be undefined?

Just asking because generating a UUID in this place can be confusing. It should come from the request itself, otherwise we risk to have different UUID generated in different places -- like, this has taken me some time to debug when working on `/debug-network` and found out MiniOxygen wasn't adding ids properly.

If it is required then it makes sense to keep it I guess.